apis/apps/v1alpha1/clusterdefinition_types.go

```diff
@@ -64,8 +65,93 @@ type ClusterDefinitionSpec struct {
 	//    `{"name": "mysql", "targetPort": "mysqlContainerPort", "port": 3306}`, and `$(SVC_PORT_mysql)` in the
 	//    connection credential value is 3306.
 	//
+	// +kubebuilder:deprecatedversion:warning="This field has been deprecated since 0.9.0"
 	// +optional
 	ConnectionCredential map[string]string `json:"connectionCredential,omitempty"`
+
+	// Topologies represents the different topologies within the cluster.
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=128
+	// +optional
+	Topologies []ClusterTopology `json:"topologies,omitempty"`
+}
+
+// ClusterTopology represents the definition for a specific cluster topology.
+type ClusterTopology struct {
+	// Name is the unique identifier for the cluster topology.
+	// Cannot be updated.
+	//
+	// +kubebuilder:validation:MaxLength=32
+	// +required
```

Comment:
The required tags on the API should be kept consistent, such as uniformly using: +kubebuilder:validation:Required.

Reply:
fixed